integration-tests/src/test/java/tachyon/client/RemoteBlockInStreamIntegrationTest.java

```diff
@@ -550,4 +552,32 @@ public void seekAroundLocalBlock() throws IOException, TException {
     Assert.assertEquals(99, is.read());
     is.close();
   }
+
+  /**
+   * Test remote read stream lock in <code>RemoteBlockInStream</code>.
+   */
+  // TODO: figure out why this test fails
+  @Test
+  public void remoteReadLockTest()
+      throws IOException, InterruptedException, FileDoesNotExistException {
+    String uniqPath = PathUtils.uniqPath();
+    for (int k = MIN_LEN + DELTA; k <= MAX_LEN; k += DELTA) {
+      TachyonFile f =
+          TachyonFSTestUtils.createByteFile(mTfs, uniqPath + "/file_" + k, mWriteTachyon, k);
+
+      long blockId = mTfs.getInfo(f).getBlockIds().get(0);
+      BlockInfo info = TachyonBlockStore.get().getInfo(blockId);
+      RemoteBlockInStream is =
+          new RemoteBlockInStream(info.getBlockId(), info.getLength(), info.getLocations().get(0)
+              .getWorkerAddress());
+      Assert.assertEquals(0, is.read());
+      mTfs.delete(f);
+      Thread.sleep(1000);
+      Assert.assertNull(mTfs.getInfo(f));
+      Assert.assertEquals(1, is.read());
```

Comment:
you may want to test reading at least more than the read buffer size, otherwise the data may not require a remote read to get.